### ISSUE TITLE
@AfterLoad() called twice

### ISSUE THREAD

**Issue by douggr** [typeorm/typeorm#2873]
**Issue type:**

[x] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I'm not sure if this is a bug or if I'm missing something, to reproduce create a simple entity:

```ts
import { AfterLoad, BaseEntity, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity("user")
export class User {
  @PrimaryGeneratedColumn("uuid")
  public readonly id: string;

  @AfterLoad()
  async $$afterLoad() {
    console.log("$$afterLoad");
  }
}
```

Call it:
```ts
return getRepository(UserModel).find().then(([ rows, count ]) => {
  res.jsonp({
    count,
    data: rows,
  });
});
```
In terminal, we can see $$afterLoad been called twice:

```shell
query: SELECT "User"."id" AS "UserModel_id" FROM "public"."user" "User"
$$afterLoad
$$afterLoad
```
Is this the expected behavior? Docs says `After load event is executed after entity has been loaded from the database.`

**Comment by csuich2** [comment]
I suspect there are 2 users in the DB. afterLoad is called once *per* entity that is loaded.

If you are looking for an afterLoad subscriber method that is invoked once for all loaded entities, it doesn't currently exist. However, I opened an issue for discussion (https://github.com/typeorm/typeorm/issues/2700) and I've attempted to implement the feature (https://github.com/typeorm/typeorm/pull/2881).
